core/app/models/spree/order/address_book.rb

```diff
@@ -55,31 +55,17 @@ def ship_address_attributes=(attributes)
       def update_or_create_address(attributes = {})
         return if attributes.blank?
 
-        attributes = attributes.select { |_k, v| v.present? }
+        attributes.transform_values! { |v| v == '' ? nil : v }
 
-        if user
-          address = user.addresses.build(attributes.except(:id)).check
-          return address if address.id
-        end
-
-        if attributes[:id]
-          address = Spree::Address.find(attributes[:id])
-          attributes.delete(:id)
+        default_address = ::Spree::Address.find_by(id: attributes[:id])
```

Comment:
@KacperMekarski if `user` is present we should only operate on `user.addresses` to avoid any permission/security issues altogether

Reply:
@damianlegawiec done.